### PR TITLE
MRG : better detect type of src for making an stc

### LINKED ISSUE #5224: problem in visualising the inverse source solution when more than one volume labels are used
How can the code can be modified to visualize the inverse solution of volume source space on MRI
 when two volume labels are given. 
For example in the code give: 
labels_vol = ['Left-Hippocampus', 'Right-Hippocampus']

[visualization of volume source space on MRI.docx](https://github.com/mne-tools/mne-python/files/2028419/visualization.of.volume.source.space.on.MRI.docx)

### PR DESCRIPTION
**Pull request by agramfort**
closes #5224 

to debug

```
import os.path as op
import matplotlib.pyplot as plt
import mne
from mne.datasets import sample
from mne import setup_volume_source_space
from mne import make_forward_solution
from mne.minimum_norm import make_inverse_operator, apply_inverse
from nilearn.plotting import plot_stat_map
from nilearn.image import index_img

# Set dir
data_path = sample.data_path()
subject = 'sample'
data_dir = op.join(data_path, 'MEG', subject)
subjects_dir = op.join(data_path, 'subjects')
bem_dir = op.join(subjects_dir, subject, 'bem')
# Set file names
fname_aseg = op.join(subjects_dir, subject, 'mri', 'aseg.mgz')
fname_model = op.join(bem_dir, '%s-5120-bem.fif' % subject)
fname_bem = op.join(bem_dir, '%s-5120-bem-sol.fif' % subject)
fname_evoked = data_dir + '/sample_audvis-ave.fif'
fname_trans = data_dir + '/sample_audvis_raw-trans.fif'
fname_fwd = data_dir + '/sample_audvis-meg-oct-6-mixed-fwd.fif'
fname_cov = data_dir + '/sample_audvis-shrunk-cov.fif'
#########################################################################
labels_vol = ['Left-Hippocampus']  # labels_vol = ['Left-Hippocampus', ‘Right-Hippocampus’]
labels_vol = ['Left-Hippocampus', 'Right-Hippocampus']
sphere = (0, 0, 0, 120)
src = setup_volume_source_space(subject, mri=fname_aseg, pos=2.0, sphere=sphere,
                                        volume_label=labels_vol, subjects_dir=subjects_dir, 
                                        verbose=True)
fwd = make_forward_solution(fname_evoked, fname_trans, src, fname_bem, mindist=5.0,  
                            meg=True, eeg=False, n_jobs=1)
src_fwd = fwd['src']
n = sum(src_fwd[i]['nuse'] for i in range(len(src_fwd)))
print('the fwd src space contains %d spaces and %d points' % (len(src_fwd), n))
# Compute inverse solution 
snr = 3.0           
inv_method = 'MNE' # or MNE
lambda2 = 1.0 / snr ** 2
# Load data
condition = 'Left Auditory'
evoked = mne.read_evokeds(fname_evoked, condition=condition,
                          baseline=(None, 0))
noise_cov = mne.read_cov(fname_cov)
# Compute inverse operator
inverse_operator = make_inverse_operator(evoked.info, fwd, noise_cov,
                                         depth=None, fixed=False)
stcs = apply_inverse(evoked, inverse_operator, lambda2, inv_method,
                     pick_ori=None)
######################################################################
src = fwd['src']
img = stcs.as_volume(src, mri_resolution=False)  # set True for full MRI resolution
t1_fname = data_path + '/subjects/sample/mri/T1.mgz'
# Plotting with nilearn ######################################################
plot_stat_map(index_img(img, 200), t1_fname, threshold=2*1e-9,
              title='%s (t=%.1f s.)' % (inv_method, stcs.times[61]))
plt.show()
```

### CODE REVIEW

**Comment by agramfort** [comment]
this is still broken but currently the _make_stc function is broken if you have a volume source space with 2 labels. If it sees 2 labels it thinks in the cortex and breaks later.

@wmvanvliet do you have time to look?

**Comment by agramfort** [comment]
I just started this to better explain the pb but it's unlikely I will be able to fix in the next few dates :(

**Comment by larsoner** [comment]
I can take over if you want. It looks like you are 90% of the way there anyway, agreed?

Also for this viz of mixed source estimates I should probably finish #4496 then by default we can have a `stc.plot` that does not require two separate visualizations.

File: mne/beamformer/_dics.py [hunk 1]
```diff
@@ -567,6 +568,7 @@ def apply_dics_csd(csd, filters, verbose=None):
 
     logger.info('[done]')
 
+    # XXX we should pass src to _make_stc
```
Comment (agramfort) [inline mne/beamformer/_dics.py]: @wmvanvliet @britta-wstnr the beamformer functions only use vertices but it's not enough to know what kind of source space you work with. There is presently an ambiguity between a standard source space with 2 hemispheres and a volume source space made of 2 labels.

do you have time to take a look?

File: mne/source_estimate.py [hunk 2]
```diff
@@ -3078,9 +3079,43 @@ def _get_ico_tris(grade, verbose=None, return_surf=False):
...
+    img : instance Nifti1Image
+        The image object.
+    """
+    return _save_stc_as_volume(fname, stc, src, dest='mri',
+                               mri_resolution=False)
+
+
+def _save_stc_as_volume(fname, stc, src, dest='mri', mri_resolution=False):
+    """Save a volume source estimate in a NIfTI file.
+
```
Comment (agramfort) [inline mne/source_estimate.py]: @britta-wstnr @wmvanvliet @larsoner ok for you to deprecate this? it was really weird to have a function called save_stc_as_volume that was always used to output the image and not save anything to disk. Besides you can do this with stc.as_volume or stc.save_as_volume.

File: mne/source_estimate.py [hunk 3]
```diff
@@ -359,10 +360,33 @@ def read_source_estimate(fname, subject=None):
...
+        warn_("src should not be None for have a robust guess of stc type.")
+        if isinstance(vertices, list) and len(vertices) == 2:
+            src_type = 'surf'
+        elif isinstance(vertices, np.ndarray) or isinstance(vertices, list)\
+                and len(vertices) == 1:
+            src_type = 'vol'
+        elif isinstance(vertices, list) and len(vertices) > 2:
+            src_type = 'mixed'
+    else:
+        src_types = set([ss['type'] for ss in src])
```
Comment (larsoner) [inline mne/source_estimate.py]: this should just be `src.kind`

https://github.com/mne-tools/mne-python/blob/master/mne/source_space.py#L207

File: mne/source_estimate.py [hunk 4]
```diff
@@ -3056,9 +3079,43 @@ def _get_ico_tris(grade, verbose=None, return_surf=False):
         return ico
 
 
+@deprecated("This function is deprecated and will be removed in version 0.18. "
+            "Use instead stc.as_volume or stc.save_as_volume methods.")
 def save_stc_as_volume(fname, stc, src, dest='mri', mri_resolution=False):
     """Save a volume source estimate in a NIfTI file.
 
+    This function is DEPRECATED.
```
Comment (larsoner) [inline mne/source_estimate.py]: You shouldn't need to add this because the decorator does something similar IIRC

File: mne/source_estimate.py [hunk 5]
```diff
@@ -3101,12 +3162,21 @@ def save_stc_as_volume(fname, stc, src, dest='mri', mri_resolution=False):
         mri_vol = np.zeros(mri_shape)
         interpolator = src[0]['interpolator']
 
-    for k, v in enumerate(vol):
-        v[mask3d] = stc.data[:, k]
-        if mri_resolution:
-            mri_vol[k] = (interpolator * v.ravel()).reshape(mri_shape3d)
+    n_vertices_seen = 0
+    for src_idx in range(len(src)):
```
Comment (larsoner) [inline mne/source_estimate.py]: `src_idx` is only used to iterate over `src` so you should just iterate directly like:
```
for src_part in src:
    assert tuple(src_part['shape'] ...)
```

File: mne/source_estimate.py (hunk 3 above)
Comment (agramfort) [reply mne/source_estimate.py]: so you suggest to change src.kind to 'mixed' and not 'combined' ? that means a deprecation cycle ...

Comment (larsoner) [reply mne/source_estimate.py]: I think it's a bug it was ever `combined`

**Comment by agramfort** [comment]
you suggest to brutally rename combined to mixed?

**Comment by larsoner** [comment]
Yes, I think `combined` was an error. And I suspect relevant workarounds should be fairly straightforward, and rarely needed (as this was likely very rarely used).

**Comment by agramfort** [comment]
ok renaming done

File: mne/source_estimate.py (hunk 2 above)
Comment (larsoner) [reply mne/source_estimate.py]: Yes okay with me

**Comment by larsoner** [comment]
LGTM other than the remaining beamformer issue. @agramfort okay to address that separately?

**Comment by agramfort** [comment]
yes I'll let @britta-wstnr take a look at the beamformer issue.

**Comment by larsoner** [comment]
Thanks @agramfort

**Pull request merged**